### PR TITLE
Send Slack ping when PR is marked ready for review

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR extends the Slack workflow to also run when a draft PR is marked ready for review.

## Testing Instructions

GitHub Actions are very hard to test. Reading the code very carefully is the only way. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/pr_ping.yml [hunk 1]
```diff
@@ -5,26 +5,40 @@ on:
...
+      pr_title: ${{ github.event.pull_request.title }}
+      pr_author: ${{ github.event.pull_request.user.login }}
+      pr_repo: ${{ github.event.pull_request.base.repo.full_name }}
     steps:
-    - name: Send Slack notification
+    - name: Send notification for new PR
+      if: github.event.action == 'opened'
+      env:
+        pr_status: ${{ github.event.pull_request.draft && ' draft ' || ' ' }}
+        pr_icon: ${{ github.event.pull_request.draft && ':pr-draft:' || ':pull-request:' }}
```
Comment (dhruvkb) [inline .github/workflows/pr_ping.yml:29]: [Reference](https://github.community/t/possible-to-use-conditional-in-the-env-section-of-a-job/135170/2); alternatively, we can also set env vars [using a `run` step](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#setting-an-environment-variable)

File: .github/workflows/pr_ping.yml [hunk 2]
```diff
@@ -5,26 +5,40 @@ on:
...
+      slack_url: ${{ secrets.SLACK_WEBHOOK_URL }}
+      pr_url: ${{ github.event.pull_request.html_url }}
+      pr_number: ${{ github.event.pull_request.number }}
+      pr_title: ${{ github.event.pull_request.title }}
+      pr_author: ${{ github.event.pull_request.user.login }}
+      pr_repo: ${{ github.event.pull_request.base.repo.full_name }}
     steps:
-    - name: Send Slack notification
+    - name: Send notification for new PR
+      if: github.event.action == 'opened'
```
Comment (dhruvkb) [inline .github/workflows/pr_ping.yml:26]: [Event payload](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request)

**Review by dhruvkb — COMMENTED** [review]
Since we cannot test, here are some references:

**Review by sarayourfriend — APPROVED** [review]
🪄 Incredible!

**Review by zackkrida — APPROVED** [review]
LGTM!

**Pull request merged**